substrate/frame/revive/src/lib.rs

```diff
@@ -1898,6 +1915,25 @@ impl<T: Config> Pallet<T> {
 			.map(|code| code.into())
 			.unwrap_or_default()
 	}
+
+	/// Returns true if unlimited contract size is allowed.
+	pub fn is_unlimited_contract_size_allowed() -> bool {
+		T::DebugEnabled::get() &&
+			DebugSettingsOf::<T>::get().unwrap_or_default().allow_unlimited_contract_size
+	}
+
+	/// Set the debug settings for the pallet.
+	pub fn set_debug_settings(settings: &Option<DebugSettings>) {
+		if let Some(settings) = settings {
+			DebugSettingsOf::<T>::put(settings);
+			if !T::DebugEnabled::get() {
+				log::warn!(
+					target: crate::LOG_TARGET,
+					"Debug settings changed, but debug features are disabled in the runtime configuration."
+				);
+			}
+		}
+	}
```

Comment:
Please don't pollute the global scope with debug functionality. Let's move to the to be created `debug` module. Maybe as an inherent function.